### DISCUSSION TITLE
How to create swarm with Gossipsub behaviour which blocks anybody except specified list of peers?

### DISCUSSION THREAD

**Discussion by Velnbur** [libp2p/rust-libp2p#5782] [Q&A] [answered]
As I understand there is only one whitelist option available, and only for topics. But how I can do the same for Peer Ids without implementing it by myself by handling these "connection" events and immediately disconnecting them?

**Comment by elenaf9** [comment, marked as answer]
There is an independent, more generic network behavior [`allow_block_list`](https://docs.rs/libp2p/latest/libp2p/allow_block_list/index.html), that only permits connections to whitelisted peers. Does that help?

**Reply by Velnbur** [reply]
Thank you!

**Comment by drHuangMHT** [comment]
If you simply want to *disconnect* from the peers that you have no interest with, use `libp2p-allow-block-list`, or try `libp2p-pnet`. But if you just want to skip the negotiation of the protocol you'll need to tinker with your NetworkBehaviour, likely with your own implementation of `NetworkBehaviour` trait without the derive macro.
